packages/esm-ward-app/src/ward-workspace/admit-patient-form-workspace/admit-patient-form-workspace.workspace.tsx

```diff
@@ -0,0 +1,174 @@
+  const { admissionEncounterTypeUuid } = useConfig<WardConfigObject>();
+  const [showErrorNotifications, setShowErrorNotifications] = useState(false);
+  const { isLoading, admissionLocation, mutate: mutateAdmissionLocation } = useAdmissionLocation();
+  const beds = useMemo(() => (isLoading ? [] : filterBeds(admissionLocation)), [admissionLocation]);
+  const getBedRepresentation = useCallback((bedLayout: BedLayout) => {
+    const bedNumber = bedLayout.bedNumber;
+    const patients =
+      bedLayout.patients.length === 0
+        ? [t('emptyBed', 'Empty')]
+        : bedLayout.patients.map((patient) => patient?.person?.preferredName?.display);
+    return [bedNumber, ...patients].join(' · ');
+  }, []);
+
+  const zodSchema = useMemo(
+    () =>
+      z.object({
+        bedId: z.number({
+          required_error: t('bedSelectionRequired', 'Please select a bed to move forward'),
+        }),
+      }),
+    [],
+  );
+
+  type FormValues = z.infer<typeof zodSchema>;
+
+  const {
+    control,
+    formState: { errors, isDirty },
+    handleSubmit,
+    getValues,
+    watch,
+  } = useForm<FormValues>({
+    resolver: zodResolver(zodSchema),
+  });
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty]);
+
+  const onSubmit = useCallback(
+    (values: FormValues) => {
+      setShowErrorNotifications(false);
+      setIsSubmitting(true);
+      const bedSelected = beds.find((bed) => bed.bedId === values.bedId);
+      createEncounter(patient.uuid, admissionEncounterTypeUuid, location?.uuid)
+        .then(async (response) => {
+          if (response.ok) {
+            return assignPatientToBed(values.bedId, patient.uuid, response.data.uuid);
+          }
+        })
+        .then(() => {
+          setIsSubmitting(false);
+          showSnackbar({
+            kind: 'success',
+            title: t('patientAdmittedSuccessfully', 'Patient admitted successfully'),
+            subtitle: t(
+              'patientAdmittedSuccessfullySubtitle',
+              '{{patientName}} has been successfully admitted to the bed {{bedNumber}}',
+              {
+                patientName: patient.person.preferredName.display,
+                bedNumber: bedSelected.bedNumber,
+              },
+            ),
+          });
+          mutateAdmissionLocation();
+          mutateInpatientRequest();
+          closeWorkspaceWithSavedChanges();
+        });
+    },
+    [beds, patient, admissionEncounterTypeUuid, location, closeWorkspaceWithSavedChanges],
+  );
+
+  const onError = useCallback((values) => {
+    setShowErrorNotifications(true);
+    setIsSubmitting(false);
+  }, []);
+
+  return (
+    <Form control={control} className={styles.form} onSubmit={handleSubmit(onSubmit, onError)}>
+      <div className={styles.formContent}>
+        <Row>
+          <Column>
+            <h2 className={styles.productiveHeading02}>{t('selectABed', 'Select a bed')}</h2>
```

Comment:
We should only display the select a bed option if the bed management module is enabled, which we can check with this feature flag it appears:

https://github.com/openmrs/openmrs-esm-patient-management/blob/main/packages/esm-ward-app/src/ward-view/ward-view.component.tsx#L20

We also should hide this selector if there are no beds configured for the current ward location

Reply:
Hi @mogoodrich!
If we don't have any beds configured, or the bed management module is not present, the user will not be able to land on this form.

Reply:
fyi for anyone else reviewing this, we discussed this on standup and clarified.